base/context.go

```diff
@@ -58,13 +64,14 @@ type Context struct {
 
 // InitDefaults sets up the default values for a context.
 func (ctx *Context) InitDefaults() {
+	ctx.Insecure = defaultInsecure
 	ctx.Certs = defaultCertsDir
 }
 
 // SSLWanted returns whether or not SSL is wanted.
-// This is indicated by the -certs flags (wanted if not empty).
+// This is indicated by the -insecure flag.
 func (ctx *Context) SSLWanted() bool {
```

Comment:
nit: Is this accessor needed?  Seems like you could just use !ctx.Insecure instead.